yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClusterSchedulerBackend.scala

```diff
@@ -37,9 +38,12 @@ private[spark] class YarnClusterSchedulerBackend(
     scheduler: TaskSchedulerImpl,
     sc: SparkContext)
   extends YarnSchedulerBackend(scheduler, sc) {
+  private val services: YarnExtensionServices = new YarnExtensionServices()
```

Comment:
It seems like `YarnSchedulerBackend` is only used in the yarn module anyway, so it could be moved there. It doesn't really make a lot of sense in `core/`.